### ISSUE TITLE
Add error message when using semaphore key < 1

### ISSUE THREAD

**Issue by ilario** [osvenskan/sysv_ipc#19]
I didn't read carefully enough the documentation and I was using a semaphore with key = 0.
The semaphore simply does nothing, neither works nor throws an error.
I would have expected an error message.

**Comment by osvenskan** [comment]
Hi @ilario, sorry, that was probably a frustrating experience. Thanks for the suggestion. Unfortunately I can't disallow a key of 0 because it might be valid on some systems.

In fact, this module used to not accept a key < 1, and I changed that 14 years ago based on a request from someone using a system where 0 can be a key. (See https://semanchuk.com/philip/sysv_ipc/history.html#v0_5_2)

I wrote a very long comment about how `key_t` is specified to be "arithmetic" (https://github.com/osvenskan/sysv_ipc/blob/develop/common.h#L57), but nothing about the specification says that a key of 0 is specifically invalid. Apparently it is on your system, but it might not be everywhere.
